src/main/java/com/redhat/devtools/lsp4ij/client/features/LSPClientFeatures.java

```diff
@@ -1105,91 +1140,74 @@ public void setServerCapabilities(@NotNull ServerCapabilities serverCapabilities
 
     @Nullable
     public TextDocumentServerCapabilityRegistry<? extends TextDocumentRegistrationOptions> getCapabilityRegistry(String method) {
-        if (LSPRequestConstants.TEXT_DOCUMENT_CODE_ACTION.equals(method)) {
-            // register 'textDocument/codeAction' capability
-            return getCodeActionFeature().getCodeActionCapabilityRegistry();
+        if (method == null) {

```

Comment:
This is the result of changing this to use `switch` which should use a jump table instead of having to evaluate each and every condition.